radixdlt/src/main/java/com/radixdlt/consensus/sync/SyncManager.java

```diff
@@ -0,0 +1,150 @@
+package com.radixdlt.consensus.sync;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.TreeSet;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.function.LongConsumer;
+import java.util.function.LongSupplier;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.radixdlt.middleware2.CommittedAtom;
+import com.radixdlt.utils.ThreadFactories;
+
+public class SyncManager {
+
+	private static final int MAX_REQUESTS_TO_SEND = 20;
+	private final Consumer<CommittedAtom> atomProcessor;
+	private final LongSupplier versionProvider;
+	private final int batchSize;
+	private final int maxAtomsQueueSize;
+	private final long patience;
+
+	private final TreeSet<CommittedAtom> commitedAtoms = new TreeSet<>((a1, a2) -> Long
+			.compare(a1.getVertexMetadata().getStateVersion(), a2.getVertexMetadata().getStateVersion()));
+	private final ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor(ThreadFactories.daemonThreads("SyncManager"));
+
+	private LongConsumer requestSender;
+	private long targetVersion = -1;
+	private ScheduledFuture<?> timeoutChecker;
+	private LongConsumer onTarget = target -> { };
+	private LongConsumer onVersion = version -> { };
+
+	public SyncManager(Consumer<CommittedAtom> atomProcessor, LongSupplier versionProvider, int batchSize,
+			long patience) {
+		this.atomProcessor = atomProcessor;
+		this.versionProvider = versionProvider;
+		this.batchSize = batchSize;
+		this.patience = patience;
+		this.maxAtomsQueueSize = MAX_REQUESTS_TO_SEND * batchSize * 2;
+	}
+
+	public void syncToVersion(long targetVersion, LongConsumer requestSender) {
+		executorService.execute(() -> {
+			if (targetVersion <= this.targetVersion) {
+				return;
+			}
+			long crtVersion = versionProvider.getAsLong();
+			if (crtVersion >= targetVersion) {
+				return;
+			}
+			this.targetVersion = targetVersion;
```

Comment:
It seems like these should be able to be passed into `sendSyncRequests()` rather than stored in instance fields.

Reply:
Not 100% convinced that targetVersion is required as a class field, but requestSender is only used in sendSyncRequests, so the class field isn't required.

Reply:
I'll change. The reason I made it a field, was for having a chance to use an up to date list of nodes, when resend the sync requests, if someone, had come with a newer list of nodes.